### PR TITLE
added basket to menu-page context

### PR DESCRIPTION
**Pull request by mplayer78**
currently logic is not extracted into useBasket hook but instead stores state in a context object on the menu-page.

### CODE REVIEW

File: chowpronto_frontend/src/shared_components/ChowButton.js [hunk 1]
```diff
@@ -4,16 +4,7 @@ import PropTypes from "prop-types";
...
-    <StyledButton
-      onClick={onClick}
-      primary={primary}
-      secondary={secondary}
-      tertiary={tertiary}
-    >
-      {title}
-    </StyledButton>
-  );
+  return <StyledButton {...props}>{title}</StyledButton>;
```
Comment (fletcher-greg) [inline chowpronto_frontend/src/shared_components/ChowButton.js:7]: Is the destructuring of onClick, primary, etc. for later?

Comment (mplayer78) [reply chowpronto_frontend/src/shared_components/ChowButton.js:7]: Yes greg, this is a bit of a weird thing with using SC but I have switched over to passing {...props} on newer branch (add-place-order-button) which I will push later today.

Comment (fletcher-greg) [reply chowpronto_frontend/src/shared_components/ChowButton.js:7]: No worries 😄

File: chowpronto_frontend/src/patron/pages/menu/components/MenuItem.js [hunk 2]
```diff
@@ -22,7 +23,7 @@ const MenuItem = (props) => {
         <HeaderRight></HeaderRight>
       </FooterBar>
       <ClearButton
-        onClick={() => updateBasket({ type: "add_item", item: props })}
+        onClick={() => dispatch({ type: "add_item_to_basket", item: props })}
```
Comment (kosenchiha) [inline chowpronto_frontend/src/patron/pages/menu/components/MenuItem.js:26]: I was mislead by ClearButton name :). This button add item to the basket and I thought it was clearing(removing) something. :)

**Pull request merged**